src/Bridge/Symfony/Validator/Validator.php

```diff
@@ -50,6 +50,10 @@ public function validate($data, array $context = [])
                 ($service = $this->container->get($validationGroups)) &&
                 \is_callable($service)
             ) {
+                if (!$service instanceof ValidationGroupsGeneratorInterface) {
```

Comment:
Is this deprecation mandatory?

Reply:
Yes, so that we can inject a service locator in 3.0

Reply:
@dunglas https://github.com/api-platform/core/pull/3346#discussion_r369280721